events.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"github.com/veandco/go-sdl2/sdl"
+	"time"
+)
+
+type SdlEventStream struct {
+	closing chan chan error
+	events  chan sdl.Event
+}
+
+func (ses *SdlEventStream) Close() error {
+	c := make(chan error)
+	ses.closing <- c
+	return <-c
+}
+
+func (ses *SdlEventStream) loop() {
+	var err error
+	for {
+		evt := sdl.PollEvent()
+		select {
+		case errc := <-ses.closing:
+			errc <- err
+			close(ses.events)
+			return
+		case <-time.After(1 * time.Second):
+		default:
```

Comment:
limit event poll frequency (e.g. 50ms), because otherwise this will keep one core completely busy.